lib/cancelled_items_retriever.py

```diff
@@ -46,40 +54,60 @@ def get_cancelled_items(self) -> Dict[str, List[str]]:
 
     return result
 
-  def get_all_email_ids(self, mail) -> Set[str]:
-    subject_searches = [[
-        "Successful cancellation of", "from your Amazon.com order"
-    ], ["Partial item(s) cancellation from your Amazon.com order"]]
-    result_ids = set()
-    for search_terms in subject_searches:
-      search_terms = ['(SUBJECT "%s")' % phrase for phrase in search_terms]
+  def get_all_email_ids(self, mail) -> Dict[str, Tuple[CancFmt, bool]]:
+    # For multi-part subject searches, | is used as delimiter because lists can't be
+    # dictionary keys in Python.
+    subject_searches = {
+        "Successful cancellation of|from your Amazon.com order": (CancFmt.VOLUNTARY, True),
```

Comment:
should we document / enum-ize the boolean here to signify that it refers to whether or not we can get quantity info from the email?

Reply:
Done.